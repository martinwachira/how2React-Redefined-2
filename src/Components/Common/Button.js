import React from "react";
import classes from "./Button.module.css";

const Button = ({ label }) => {
  return (
    <div>
      <button data-testid="btn" className={classes.button}>
        {label}
      </button>
    </div>
  );
};

export default Button;
