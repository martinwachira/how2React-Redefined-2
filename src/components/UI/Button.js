import React from "react";
import classes from "./Button.module.css";

const Button = (props) => {
  return (
    <div>
      <button type="submit" className={classes.button}>
        Add User
      </button>
    </div>
  );
};

export default Button;
