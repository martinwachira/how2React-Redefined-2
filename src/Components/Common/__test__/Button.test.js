import "@testing-library/jest-dom/extend-expect";

import { cleanup, render } from "@testing-library/react";

import Button from "../Button";
import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";

afterEach(cleanup);
it("renders without fail", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Button></Button>, div);
});

it("renders btn correctly", () => {
  const { getByTestId } = render(<Button label="click"></Button>);
  expect(getByTestId("btn")).toHaveTextContent("click");
});

it("renders btn 2 correctly", () => {
  const { getByTestId } = render(<Button label="order"></Button>);
  expect(getByTestId("btn")).toHaveTextContent("order");
});

it("matches snapshot", () => {
  const tree = renderer.create(<Button label="order"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});

it("matches snapshot 2", () => {
  const tree = renderer.create(<Button label="order 2"></Button>).toJSON();
  expect(tree).toMatchSnapshot();
});
