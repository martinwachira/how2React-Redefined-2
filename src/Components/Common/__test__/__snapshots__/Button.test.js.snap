// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <button
    className="button"
    data-testid="btn"
  >
    order
  </button>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div>
  <button
    className="button"
    data-testid="btn"
  >
    order 2
  </button>
</div>
`;
